### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by pauldariye**
Update installation instruction for Typescript

**What**:
- Documentation

**Why**:
- Update instruction for Typescript

**How**:

**Checklist**:

- [x] Documentation
- [ ] Tests
- [ ] Updated Type Definitions
- [ ] Ready to be merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,9 +106,8 @@ import {toBeInTheDocument, toHaveClass} from '@testing-library/jest-dom'
 expect.extend({toBeInTheDocument, toHaveClass})
 ```
 
-> Note: when using TypeScript, this way of importing matchers won't provide the
-> necessary type definitions. More on this
-> [here](https://github.com/testing-library/jest-dom/pull/11#issuecomment-387817459).
+**Note** If you're using Typescript, make sure your setup file is a `.ts` and not a `.js` to include the necessary types.
```
Comment (gnapse) [inline README.md]: Nit pick:

```suggestion
**Note** If you're using TypeScript, make sure your setup file is a `.ts` and not a `.js` to include the necessary types.
```

Comment (gnapse) [reply README.md]: Also, an extra nit-pick: this kind of paragraphs that are meant as a side note in the README, and not part of the normal flow of reading, I'd rather have them be a note-like paragraph, preceding in each line with markdown notation `>`, just as the paragraph you replaced originally.

**Comment by pauldariye** [comment]
> Thanks for your contribution. Indeed, there are things to be improved about this TypeScript aspect in the README. However, I'm not entirely sure this replacement is accurate with respect to the intentions of the text you're replacing.
> 
> The text being replaced was meant to convey the fact that picking and choosing to import just some of the custom matchers, and extent jest just with the ones you want, and not all, this won't make TypeScript happy. And using a setup file in TS as you mention won't fix that either.
> 
> Using a TS setup file will fix importing jest-dom in the usual manner. So I think the text you have included here, rather than replacing this part of the README, should be inserted, without replacing anything, before the paragraph in the README a few lines above this part, the one that starts with the sentence "Alternatively, you can selectively import only..."
> 
> Can you comment on these concerns of mine and clarify if I'm misreading something here?
> 
> (Also, as a side note, this PR is related to #123 so I'll mention that here to create the cross reference to that issue, which, when solved, may even change this update).

Hey @gnapse, I think you're right. I misread that. I'll put back that paragraph.

**Comment by pauldariye** [comment]
I've updated per your comment and suggestion @gnapse and thanks!

**Pull request merged**